nextcord/application_command.py

```diff
@@ -0,0 +1,1266 @@
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+import asyncio
+from inspect import signature, Parameter
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Set,
+    TYPE_CHECKING,
+    Tuple,
+    Type,
+    Union,
+)
+
+from . import utils
+from .abc import GuildChannel
+from .enums import ApplicationCommandType, ApplicationCommandOptionType, ChannelType
+from .interactions import Interaction
+from .guild import Guild
+from .member import Member
+from .message import Message
+from .mixins import Hashable
+from .role import Role
+from .user import User
+from .utils import MISSING
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+
+
+__all__ = (
+    "ClientCog",
+    "SlashOption",
+    "ApplicationSubcommand",
+    "ApplicationCommand",
+    "CommandOption",
+    "slash_command",
+    "message_command",
+    "user_command",
+)
+
+
+class InvalidCommandType(Exception):
+    """Raised when an unhandled Application Command type is encountered."""
+    pass
+
+
+class ClientCog:
+    # TODO: I get it's a terrible name, I just don't want it to duplicate current Cog right now.
+    __cog_application_commands__: Dict[int, ApplicationCommand]
+    __cog_to_register__: List[ApplicationCommand]
+
+    def __new__(cls, *args: Any, **kwargs: Any):
+        new_cls = super(ClientCog, cls).__new__(cls)
+        new_cls._read_methods()
+        return new_cls
+
+    def _read_methods(self) -> None:
+        self.__cog_to_register__ = []
+        for base in reversed(self.__class__.__mro__):
+            for elem, value in base.__dict__.items():
+                is_static_method = isinstance(value, staticmethod)
+                if is_static_method:
+                    value = value.__func__
+                if isinstance(value, ApplicationCommand):
+                    if isinstance(value, staticmethod):
+                        raise TypeError(f"Command {self.__name__}.{elem} can not be a staticmethod.")
+                    value.set_self_argument(self)
+                    self.__cog_to_register__.append(value)
+                elif isinstance(value, ApplicationSubcommand):
+                    value.set_self_argument(self)
+
+    @property
+    def to_register(self) -> List[ApplicationCommand]:
+        return self.__cog_to_register__
+
+
+class SlashOption:
+    def __init__(self, name: str = MISSING,
+                 description: str = MISSING,
+                 required: bool = MISSING,
+                 choices: Dict[str, Union[str, int, float]] = MISSING,
+                 channel_types: List[Type[ChannelType]] = MISSING,
+                 min_value: Union[int, float] = MISSING,
+                 max_value: Union[int, float] = MISSING,
+                 autocomplete: bool = MISSING,
+                 default: Optional[Any] = None,
+                 verify: bool = True
+                 ):
```

Comment:
Fix this indentation / layout to be more pep 8 inline. Deindent by 1, move self and name to under ``__init__`` and on there own line